src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -1126,6 +1130,54 @@ void CorProfiler::InternalAddInstrumentation(WCHAR* id, CallTargetDefinition* it
     }
 }
 
+void CorProfiler::ConfigureContinuousProfiler(bool         threadSamplingEnabled,
+                                              unsigned int threadSamplingInterval,
+                                              bool         allocationSamplingEnabled,
+                                              unsigned int maxMemorySamplesPerMinute)
+{
+    Logger::Info("ConfigureContinuousProfiler: thread sampling enabled: ", threadSamplingEnabled,
+                 ", thread sampling interval: ", threadSamplingInterval, ", allocationSamplingEnabled: ",
+                 allocationSamplingEnabled, ", max memory samples per minute: ", maxMemorySamplesPerMinute);
+
+    if (!threadSamplingEnabled && !allocationSamplingEnabled)
+    {
+        Logger::Debug("ConfigureContinuousProfiler: Thread sampling and allocations sampling disabled.");
+        return;
+    }
+
+    DWORD pdvEventsLow;
+    DWORD pdvEventsHigh;
+    auto  hr = this->info12_->GetEventMask2(&pdvEventsLow, &pdvEventsHigh);
+    if (FAILED(hr))
+    {
+        Logger::Warn("ConfigureContinuousProfiler: Failed to take event masks for continuous profiler.");
+        return;
+    }
+
+    pdvEventsLow |= COR_PRF_MONITOR_THREADS | COR_PRF_ENABLE_STACK_SNAPSHOT;
```

Comment:
Are these flags needed if only allocation sampling is enabled?

Reply:
Ref: https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/cor-prf-monitor-enumeration

* `COR_PRF_MONITOR_THREADS` - needed to have the info about thread events. Needed to corelate stack traces with Traces/Spans. Needed for both allocation and thread sampling
* `COR_PRF_ENABLE_STACK_SNAPSHOT` - Enables calls to the [DoStackSnapshot](https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/icorprofilerinfo2-dostacksnapshot-method) method. `DoStackSnapshot` method is also executed for thread and allocation sampling.

Reply:
Wouldn't it be worth adding a comment in code?